### PR TITLE
Configure web server base path

### PR DESCRIPTION
**Pull request by aeschli**
Adds a new server option `--server-base-path` that allows to configure the URL path under which the web server is started,

### CODE REVIEW

**Comment by vishaltak** [comment]
@aeschli - Wanted to understand if this is being worked on and if there is any target for this? Thanks for working on this.

File: .yarn/install-state.gz
Comment (Tyriar) [inline .yarn/install-state.gz:1]: What is this file about, do we need this committed?

Comment (aeschli) [reply .yarn/install-state.gz:1]: good catch. Removed it

**Review by connor4312 — APPROVED** [review]

**Comment by ab077u** [comment]
Thanks for this feature. I was trying to figure out how to make this subpath work with using nginx, until I stumbled on this.

I am trying out the following insiders build, and found some issue with themes and icon rendering. One observation is if I start the server **without** `--server-base-path` option and access it, then start with `--server-base-path` it renders properly. In addition, the welcome screen walkthroughs icons for installed extensions fail to load (screenshots below). 

```bash
code-insiders serve-web --server-base-path /foo/bar --port 8883 --host 0.0.0.0 --without-connection-token --log debug                                                      
```
image

```
Version: 1.88.0-insider (Universal)
Commit: b40321263dbe14515f436b9b34ab5c7d8d5526ae
Date: 2024-03-21T17:26:53.619Z
Electron: 28.2.6
ElectronBuildId: 27476517
Chromium: 120.0.6099.291
Node.js: 18.18.2
V8: 12.0.267.19-electron.0
OS: Darwin arm64 23.2.0
```

image

**Comment by mering** [comment]
Thanks for this great feature. Is there a way to only set it as proxy path in a sense that additional resources get loaded with this prefix but the server answers to requests without this prefix?

For example, I have a proxy which redirects "/a/b" to "code_servce_web_container:8080", so additional resources need to be loaded from "/a/b/path" while the code server should respond on "/path".

**Comment by oakaigh** [comment]
> additional resources get loaded with this prefix but the server answers to requests without this prefix

@mering Is there no way for the **proxy** server to be configured to strip the prefix? Implementing automatic prefix stripping on the code server itself can be tricky, particularly since it wasn't designed to be very flexible at path handling.

**Comment by mering** [comment]
@oakaigh The proxy server _does_ strip the prefix and this is exactly the problem:
When accessing `/prefix/devcontainer` the proxy redirects this to `vscode:port/devcontainer` which succeeds but it tries to download the resources from `/insider-ID/static/out/...` instead of `/prefix/insider-ID/static/out/...` so the proxy doesn't know where to forward the request.

Embedding would indeed solve the problem.

**Comment by mering** [comment]
@oakaigh Can you explain why relative paths don't work?
A relative path `./insider-ID/static/out/...` referenced from the main page `/` should be the same as `/insider-ID/static/out/...`, no?

**Comment by oakaigh** [comment]
@mering Apologize for the confusion. I stand corrected - actually the paths need to be passed to the server **as is**. So no, the paths cannot be stripped outside the code server, unless there's an option to enable the server to return HTMLs with stripped resource paths (so that it's informed of the stripped paths). I don't think this is currently implemented anywhere.

**Comment by aeschli** [comment]
@oakaigh @mering Can we discuss this in a new issue? 
Re missing icons: Last week I fixed and issue about icon loading. https://github.com/microsoft/vscode/issues/208359
So please make sure you test this in the latest insiders.

**Comment by mering** [comment]
@aeschli I created a new issue in #210399.

**Pull request merged**